src/controllers/cart.js

```diff
@@ -23,14 +23,14 @@ exports.postCart = async (req, res, next) => {
   }
 };
 
-exports.deleteCart = async (req, res) => {
+exports.deleteCart = async (req, res, next) => {
   try {
     Cart.deleteOne({ customerId: req.user.id });
-    res.status(204).json({
+    return res.status(204).json({
       message: 'the cart was deleted',
     });
   } catch (err) {
-    res.status(500).json({ message: 'Failed to delete cart', error: err });
+    return next(new CustomError('Failed to delete cart'));
```

Comment:
You can't tell what type of error is happening here so change 'Failed to delete cart' to err.message and add 500 status code